### PR TITLE
[gh-2022] update address data in mnemonic data.

### LINKED ISSUE #2022: [cli] `rooch account create` can not create more account
The `rooch account create` command can not create more accounts.

Execute the `rooch account create` twice. It outputs the same account address.

### PR DESCRIPTION
**Pull request by feliciss**
## Summary

Update address data in mnemonic.

See discussion in #2022 for details.

- Closes #2022

### CODE REVIEW

File: crates/rooch-key/src/keystore/account_keystore.rs [hunk 1]
```diff
@@ -43,13 +43,12 @@ pub trait AccountKeystore {
 
     fn generate_and_add_new_key(
         &mut self,
+        account_index: u32,
         password: Option<String>,
     ) -> Result<GeneratedKeyPair, anyhow::Error> {
         // load mnemonic phrase from keystore
         let mnemonic = self.get_mnemonic(password.clone())?;
-        let account_index = mnemonic.mnemonic_data.addresses.len() as u32;
```
Comment (jolestar) [inline crates/rooch-key/src/keystore/account_keystore.rs:50]: Why is the `mnemonic_data.addresses.len()` incorrect?

Comment (feliciss) [reply crates/rooch-key/src/keystore/account_keystore.rs:50]: > Why is the `mnemonic_data.addresses.len()` incorrect?

See #2022 for details.

Cause there's only one address in `mnemonic_data`.

Comment (jolestar) [reply crates/rooch-key/src/keystore/account_keystore.rs:50]: Why only one address in mnemonic_data? It also should be fixed

Comment (feliciss) [reply crates/rooch-key/src/keystore/account_keystore.rs:50]: > Why only one address in mnemonic_data? It also should be fixed

https://github.com/rooch-network/rooch/pull/950#issuecomment-1760561544

We should discuss it with @baicaiyihao

Comment (baichuan3) [reply crates/rooch-key/src/keystore/account_keystore.rs:50]: Use `keystore.addresses()` as the account index is ok. If do this, we can remove `MnemonicData.addresses` field to simplify the MnemonicData definition.
Or when creating an account, need to update the new address to `MnemonicData.addresses`, and can keep `MnemonicData.addresses`.
Both implementations can work.

Comment (feliciss) [reply crates/rooch-key/src/keystore/account_keystore.rs:50]: > Use `keystore.addresses()` as the account index is ok. If do this, we can remove `MnemonicData.addresses` field to simplify the MnemonicData definition. Or when creating an account, need to update the new address to `MnemonicData.addresses`, and can keep `MnemonicData.addresses`. Both implementations can work.

It sounds like that we can remove MnemonicData.addresses as mentioned in https://github.com/rooch-network/rooch/pull/950#issuecomment-1760542376.

Comment (feliciss) [reply crates/rooch-key/src/keystore/account_keystore.rs:50]: @jolestar

**Comment by feliciss** [comment]
I updated the process to add the Rooch addresses to mnemonic data while using the current length of keystores as account index to create keys as the keys are needed to be generated before adding to mnemonic data.

File: crates/rooch-key/src/keystore/base_keystore.rs [hunk 2]
```diff
@@ -56,6 +56,18 @@ impl AccountKeystore for BaseKeyStore {
         Ok(())
     }
 
+    fn add_addresses_to_mnemonic_data(
```
Comment (jolestar) [inline crates/rooch-key/src/keystore/base_keystore.rs:59]: 有了这个方法后，mnemonic_data 中的 address 长度就正确了，所以那个 account_index 参数就可以不需要改了。

Comment (feliciss) [reply crates/rooch-key/src/keystore/base_keystore.rs:59]: > 有了这个方法后，mnemonic_data 中的 address 长度就正确了，所以那个 account_index 参数就可以不需要改了。

ok.

**Review by jolestar — APPROVED** [review]

**Pull request merged**